---
title: "Human Microbiome Cooccurance"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Human Microbiome Cooccurance}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

Load packages
```{r, eval=FALSE}
library(phyloseq)
library(parallelDist)
library(vegan)
library(backbone)
library(tidyverse)
library(igraph)
library(visNetwork)
#library(CoreMiro)
```

```{r, eval=FALSE}
hmbp_md<-read.csv("../data-raw/v35_map_uniquebyPSN.csv", sep = ",")
rownames(hmbp_md)<-hmbp_md$X.SampleID
hmbp_md$X.SampleID = NULL
hmbp_otu<-read.csv("../data-raw/otu_table_psn_v35.txt", sep = "\t")
#end long so it grabs everything
names(hmbp_otu)<-substr(names(hmbp_otu), start = 2, stop = 20)
rownames(hmbp_otu)<-hmbp_otu$.OTU.ID
hmbp_otu$.OTU_ID = NULL
hmpb_tax<-as.character(hmbp_otu$onsensus.Lineage)
#export and split to meet structure for inclusinon in phyloseq object
```

```{r, eval=FALSE}
commonalities<-intersect(names(hmbp_otu), rownames(hmbp_md))
hmbp_otu_sub<-hmbp_otu[,names(hmbp_otu) %in% commonalities]
hmbp_md_sub<-hmbp_md[rownames(hmbp_md) %in% commonalities,]
#human_tax<-read.csv("../data-raw/Human_taxonomy.csv", header = T)
#rownames(human_tax)<-human_tax$OTU_ID
#uman_tax$OTU_ID=NULL
#human_tax<- as.matrix(human_tax)
human_tab<- otu_table(hmbp_otu_sub, taxa_are_rows = T)
human_sample_data<-sample_data(hmbp_md_sub)
human_taxonomy<-tax_table(human_tax)
human_ps<-phyloseq(human_tab, human_sample_data, human_taxonomy)
#subset to those samples that our core analysis was conducted with
hmbp_working_set<-subset_samples(physeq = human_ps, HMPbodysubsite=="Stool")
human_working_set <- prune_taxa(taxa_sums(hmbp_working_set) >=1, hmbp_working_set)
```

Assign core membership
```{r, eval=FALSE}
human_core<-core_methods(human_stool)
human_core_df<- data.frame(human_core)
table(human_core_df$name, human_core_df$value)
# Create table all taxa that are never assigned.
wide_human<-pivot_wider(human_core_df)
#table(rowSums(wide_human[,5:8]))
wide_human$num_methods<-(rowSums(wide_human[,5:8]))
#tables number assigned by each method to add in paper. 
#table(wide_human$`Proportion of Sequence Reads`)
#table(wide_human$`Proportion of Sequence Reads and Replicates`)
#table(wide_human$`Hard Cut Off`)
#table(wide_human$`Proportion of Sequence Replicates`)
#pull out those common taxa found by all
core_4<-(wide_human[wide_human$num_methods==4,])$X
#pull out those taxa included by each method
prop_seq_reads_core<-(wide_human[wide_human$`Summation of Sequence Reads`==1,])$X
prop_seq_readsnrep_core<-(wide_human[wide_human$`Proportion of Sequence Reads and Replicates`==1,])$X
HC_core<-(wide_human[wide_human$`Hard Cutoff`==1,])$X
prop_rep_core<-(wide_human[wide_human$`Proportion of Sequence Replicates`==1,])$X
```

Adonis of full human microbiome stool dataset

```{r, eval=FALSE}
md_a<-data.frame(sample_data(human))
md_a$visitno<-as.factor(md_a$visitno)
otu_a<-data.frame(otu_table(human))
otu_a_t<-data.frame(t(otu_a))
otu_a_t<-data.matrix(otu_a_t)
otu_human_dist<-parallelDist(otu_a_t, method = "bray")
adonis(otu_human_dist ~ visitno + sex + RUNCENTER, data=md_a)
otu_human_dist<-parallelDist(otu_a_t, method = "binary")
adonis(otu_human_dist ~ visitno + sex + RUNCENTER, data=md_a)
```

Core 4

```{r, eval=FALSE}
otu_a_t<-data.frame(otu_a_t)
human_core_4<- otu_a_t[,names(otu_a_t) %in% core_4]
human_core_4<-data.matrix(human_core_4)
human_core_4_dist<-parallelDist(human_core_4, method = "bray")
adonis(human_core_4_dist ~ visitno + sex + RUNCENTER, data=md_a)
human_core_4_dist<-parallelDist(human_core_4, method = "binary")
adonis(human_core_4_dist ~ visitno + sex + RUNCENTER, data=md_a)
```

Prop Seq reads

```{r, eval=FALSE}
otu_a_t<-data.frame(otu_a_t)
human_prop_seq_reads_core<- otu_a_t[,names(otu_a_t) %in% prop_seq_reads_core]
human_prop_seq_reads_core<-data.matrix(human_prop_seq_reads_core)
human_prop_seq_reads_core_dist<-parallelDist(human_prop_seq_reads_core, method = "bray")
adonis(human_prop_seq_reads_core_dist ~ visitno + sex + RUNCENTER, data=md_a)
human_prop_seq_reads_core_dist<-parallelDist(human_prop_seq_reads_core, method = "binary")
adonis(human_prop_seq_reads_core_dist ~ visitno + sex + RUNCENTER, data=md_a)
```

Prop Read N Rep
```{r, eval=FALSE}
otu_a_t<-data.frame(otu_a_t)
human_prop_rnr_core<- otu_a_t[,names(otu_a_t) %in% prop_seq_readsnrep_core]
human_prop_rnr_core<-data.matrix(human_prop_rnr_core)
human_prop_rnr_core_dist<-parallelDist(human_prop_rnr_core, method = "bray")
adonis(human_prop_rnr_core_dist ~ visitno + sex + RUNCENTER, data=md_a)
human_prop_rnr_core_dist<-parallelDist(human_prop_rnr_core, method = "binary")
adonis(human_prop_rnr_core_dist ~ visitno + sex + RUNCENTER, data=md_a)
```

HC
```{r, eval=FALSE}
otu_a_t<-data.frame(otu_a_t)
human_HC_core<- otu_a_t[,names(otu_a_t) %in% HC_core]
human_HC_core<-data.matrix(human_HC_core)
human_HC_core_dist<-parallelDist(human_HC_core, method = "bray")
adonis(human_HC_core_dist ~ visitno + sex + RUNCENTER, data=md_a)
human_HC_core_dist<-parallelDist(human_HC_core, method = "binary")
adonis(human_HC_core_dist ~ visitno + sex + RUNCENTER, data=md_a)
```

Prop Rep
```{r, eval=FALSE}
otu_a_t<-data.frame(otu_a_t)
human_prop_rep_core<- otu_a_t[,names(otu_a_t) %in% prop_rep_core]
human_prop_rep_core<-data.matrix(human_prop_rep_core)
human_prop_rep_core_dist<-parallelDist(human_prop_rep_core, method = "bray")
adonis(human_prop_rep_core_dist ~ visitno + sex + RUNCENTER, data=md_a)
human_prop_rep_core_dist<-parallelDist(human_prop_rep_core, method = "binary")
adonis(human_prop_rep_core_dist ~ visitno + sex + RUNCENTER, data=md_a)
```

#Network analysis 
```{r, eval=FALSE}
md_a<-data.frame(sample_data(human))
md_a$visitno<-as.factor(md_a$visitno)
otu_a<-data.frame(otu_table(human))
#Network of full dataset
#http://pablobarbera.com/big-data-upf/html/02b-networks-descriptive-analysis.html
network_test_dat<-otu_a
network_test_dat2 <- network_test_dat %>% mutate_if(is.numeric, ~1 * (. > 0))
rownames(network_test_dat2)<-rownames(network_test_dat)
network_human<-hyperg(as.matrix(network_test_dat2))
hyperg_human_network_sig <- backbone.extract(network_human, alpha = .0001, class = "igraph", fwer = "bonferroni", narrative = TRUE)
#Gives same output as below.
#sorted_degree_human_nodes<-sort(degree(hyperg_human_network_sig), decreasing = T)
node_extract_full_human<-hyperg_human_network_sig[[]]
nodes_with_sig_full_human<-node_extract_full_human[lapply(node_extract_full_human,length)>0]
full_human_node_w_num_edges<-sort(lengths(nodes_with_sig_full_human), decreasing = T) 
str(full_human_node_w_num_edges)
human_node_list<-names(full_human_node_w_num_edges)
```

Comparison of human nodes with taxa identified by core methods
Intersetion of core assignment methods with global signficance network nodes

```{r, eval=FALSE}
#Core_4
core_network_non_common_core4<-full_human_node_w_num_edges[!names(full_human_node_w_num_edges) %in% intersect(core_4, human_node_list)]
core_network_commanlity_core4<-full_human_node_w_num_edges[names(full_human_node_w_num_edges) %in% intersect(core_4, human_node_list)]
#Prop_Reads
core_network_non_common_prop_reads<-full_human_node_w_num_edges[!names(full_human_node_w_num_edges) %in% intersect(prop_seq_reads_core, human_node_list)]
core_network_commanlity_prop_reads<-full_human_node_w_num_edges[names(full_human_node_w_num_edges) %in% intersect(prop_seq_reads_core, human_node_list)]
#Prop_Reps
core_network_non_common_prop_reps<-full_human_node_w_num_edges[!names(full_human_node_w_num_edges) %in% intersect(prop_rep_core, human_node_list)]
core_network_commanlity_prop_reps<-full_human_node_w_num_edges[names(full_human_node_w_num_edges) %in% intersect(prop_rep_core, human_node_list)]
#PropRnR
core_network_non_common_PropRnR<-full_human_node_w_num_edges[!names(full_human_node_w_num_edges) %in% intersect(prop_seq_readsnrep_core, human_node_list)]
core_network_commanlity_Prop_RnR<-full_human_node_w_num_edges[names(full_human_node_w_num_edges) %in% intersect(prop_seq_readsnrep_core, human_node_list)]
#HCs
core_network_non_common_HC<-full_human_node_w_num_edges[!names(full_human_node_w_num_edges) %in% intersect(HC_core, human_node_list)]
core_network_commanlity_HC<-full_human_node_w_num_edges[names(full_human_node_w_num_edges) %in% intersect(HC_core, human_node_list)]
```

Barplot of inclusion by group status
```{r, eval=FALSE}
taxa_in_core_by_any_method <- unique(c(as.character(prop_seq_reads_core), as.character(prop_rep_core), as.character(prop_seq_readsnrep_core), as.character(HC_core)))
#taxa in network with edge > 1
full_human_node_w_num_edges
#could be accomplished with just the intersect command too. Its currenlty the long way. 
Shared_core_network_taxa<-full_human_node_w_num_edges[names(full_human_node_w_num_edges) %in% intersect(taxa_in_core_by_any_method, human_node_list)]
taxa_only_in_core<-taxa_in_core_by_any_method[!taxa_in_core_by_any_method %in% names(Shared_core_network_taxa)]
taxa_only_in_network<-full_human_node_w_num_edges[!names(full_human_node_w_num_edges) %in% names(Shared_core_network_taxa)] 
#summary of degree of each node. No info is available for taxa only in the core becasue they werent included in the network. 
summary(Shared_core_network_taxa)
summary(taxa_only_in_network)
# I think this would be better as a venn diagram actaully. 
counts<-(c(length(Shared_core_network_taxa), length(taxa_only_in_core), length(taxa_only_in_network)))
barplot(counts)
```