<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Analysis and Plots • CoreMicro</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Analysis and Plots">
<meta property="og:description" content="CoreMicro">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">CoreMicro</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li class="divider">
    <li class="dropdown-header">Using CoreMicro</li>
    <li>
      <a href="../articles/analysis_and_plots.html">Analysis and Plots</a>
    </li>
    <li class="divider">
    <li class="dropdown-header">Manuscript Reproducibility</li>
    <li>
      <a href="../articles/figure1.html">Figure 1</a>
    </li>
    <li>
      <a href="../articles/figure2.html">Figure 2</a>
    </li>
    <li>
      <a href="../articles/figure3.html">Figure 3</a>
    </li>
    <li>
      <a href="../articles/human_figure.html">Figure: Human Venn Diagram</a>
    </li>
    <li>
      <a href="../articles/arabidopsis_figure.html">Figure: Arabidopsis Venn Diagram</a>
    </li>
    <li>
      <a href="../articles/common-counts.html">Common Counts</a>
    </li>
    <li>
      <a href="../articles/co-occurance.html">Human Microbiome Cooccurance</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MayaGans/CoreMicro/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><link href="analysis_and_plots_files/anchor-sections-1.0/anchor-sections.css" rel="stylesheet">
<script src="analysis_and_plots_files/anchor-sections-1.0/anchor-sections.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Analysis and Plots</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MayaGans/CoreMicro/blob/master/vignettes/analysis_and_plots.Rmd"><code>vignettes/analysis_and_plots.Rmd</code></a></small>
      <div class="hidden name"><code>analysis_and_plots.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>We include two datasets with the package. One of which is the Arabidopsis dataset derived from Lundberg et al. 2012. The dataset included is a subset of the entire core <i>Arabidopsis thaliana</i> root microbiome dataset. We utilized only the rhizosphere samples from the M21 site.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">arabidopsis</span></code></pre></div>
</div>
<div id="data-structure" class="section level1">
<h1 class="hasAnchor">
<a href="#data-structure" class="anchor"></a>Data Structure</h1>
<p>By default {coremicro} assumes your data is in a wide format:</p>
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#qyinmuzzot .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#qyinmuzzot .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qyinmuzzot .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#qyinmuzzot .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#qyinmuzzot .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qyinmuzzot .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qyinmuzzot .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#qyinmuzzot .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#qyinmuzzot .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#qyinmuzzot .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#qyinmuzzot .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#qyinmuzzot .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#qyinmuzzot .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#qyinmuzzot .gt_from_md > :first-child {
  margin-top: 0;
}

#qyinmuzzot .gt_from_md > :last-child {
  margin-bottom: 0;
}

#qyinmuzzot .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#qyinmuzzot .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#qyinmuzzot .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qyinmuzzot .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#qyinmuzzot .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qyinmuzzot .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#qyinmuzzot .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#qyinmuzzot .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qyinmuzzot .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qyinmuzzot .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#qyinmuzzot .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qyinmuzzot .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#qyinmuzzot .gt_left {
  text-align: left;
}

#qyinmuzzot .gt_center {
  text-align: center;
}

#qyinmuzzot .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#qyinmuzzot .gt_font_normal {
  font-weight: normal;
}

#qyinmuzzot .gt_font_bold {
  font-weight: bold;
}

#qyinmuzzot .gt_font_italic {
  font-style: italic;
}

#qyinmuzzot .gt_super {
  font-size: 65%;
}

#qyinmuzzot .gt_footnote_marks {
  font-style: italic;
  font-size: 65%;
}
</style>
<div id="qyinmuzzot" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;"><table class="table gt_table">
<thead class="gt_col_headings"><tr>
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">X</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">site_1</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">site_2</th>
    </tr></thead>
<tbody class="gt_table_body">
<tr>
<td class="gt_row gt_left">OTU_1</td>
      <td class="gt_row gt_right">0</td>
      <td class="gt_row gt_right">20</td>
    </tr>
<tr>
<td class="gt_row gt_left">OTU_2</td>
      <td class="gt_row gt_right">10</td>
      <td class="gt_row gt_right">1</td>
    </tr>
<tr>
<td class="gt_row gt_left">OTU_3</td>
      <td class="gt_row gt_right">20</td>
      <td class="gt_row gt_right">20</td>
    </tr>
</tbody>
</table></div>
<p>Where the first column is your taxa names, and all subsequent columns are numeric containing your taxa counts. If your data is in long format you can set <code>taxa_as_rows = FALSE</code></p>
</div>
<div id="mean-variance-covariance" class="section level1">
<h1 class="hasAnchor">
<a href="#mean-variance-covariance" class="anchor"></a>Mean, Variance, Covariance</h1>
<p>The <code>summarize_taxa</code> function can be used to calculate the mean <code>Mean</code>, variance <code>Variance</code>, and coefficient of variance <code>CV</code>, for each taxa in the dataset:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/summarise_taxa.html">summarise_taxa</a></span><span class="op">(</span><span class="va">arabidopsis</span><span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 14,890 x 4
##    X            Mean Variance    CV
##    &lt;chr&gt;       &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt;
##  1 OTU_100   0.368    0.707   1.92 
##  2 OTU_1000  0.00694  0.00692 0.997
##  3 OTU_10000 0.698    5.85    8.38 
##  4 OTU_10002 0.260    0.284   1.09 
##  5 OTU_10003 0.00694  0.00692 0.997
##  6 OTU_10006 0.146    0.160   1.10 
##  7 OTU_10007 0.00694  0.00692 0.997
##  8 OTU_10008 0.0451   0.0502  1.11 
##  9 OTU_1001  0.0694   0.128   1.84 
## 10 OTU_10011 0.861    5.30    6.16 
## # … with 14,880 more rows</code></pre>
</div>
<div id="single-methods" class="section level1">
<h1 class="hasAnchor">
<a href="#single-methods" class="anchor"></a>Single Methods</h1>
<p>We can use the built in <code>arabidopsis</code> dataset to look at each of the core methods included in the package thusfar:</p>
<div id="hard-cutoff" class="section level3">
<h3 class="hasAnchor">
<a href="#hard-cutoff" class="anchor"></a>Hard Cutoff</h3>
<p>This method assigns taxa to the core if they are present in more than a pre-determined number of sites. The default threshhold is 25 counts in at least 5 sites, you can change these two parameters using the <code>cutoff</code> and <code>site</code> arguments</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/hard_cutoff.html">hard_cutoff</a></span><span class="op">(</span><span class="va">arabidopsis</span>, cutoff <span class="op">=</span> <span class="fl">25</span>, sites <span class="op">=</span> <span class="fl">5</span><span class="op">)</span></code></pre></div>
</div>
<div id="summation-of-sequence-reads" class="section level3">
<h3 class="hasAnchor">
<a href="#summation-of-sequence-reads" class="anchor"></a>Summation of Sequence Reads</h3>
<p>This method assigns taxa to the core if they are in the top X% of reads. Taxa are ranked in abundance and the cumulative sum is recorded. Any taxa which appears before some cutoff percentage is included in the core. The default is for taxa to be assigned to the core if they account for the first 75% of the reads. You can change this using the <code>nreads</code> argument</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/abundance_core.html">abundance_core</a></span><span class="op">(</span><span class="va">arabidopsis</span>, nreads <span class="op">=</span> <span class="fl">0.75</span><span class="op">)</span></code></pre></div>
</div>
<div id="proportion-of-sequence-replicates" class="section level3">
<h3 class="hasAnchor">
<a href="#proportion-of-sequence-replicates" class="anchor"></a>Proportion of Sequence Replicates</h3>
<p>This method assigns taxa to the core if they account for some proportion of the total reads for the sequencing run. As a default, taxa must be present in at least 50% of sites:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/occupancy_core.html">occupancy_core</a></span><span class="op">(</span><span class="va">arabidopsis</span>, occupancy_core <span class="op">=</span> <span class="fl">0.5</span><span class="op">)</span></code></pre></div>
</div>
<div id="proportion-of-sequence-reads-and-replicates" class="section level3">
<h3 class="hasAnchor">
<a href="#proportion-of-sequence-reads-and-replicates" class="anchor"></a>Proportion of Sequence Reads and Replicates</h3>
<p>This method assigns taxa to the core if they account for some proportion of the total reads for the sequencing run and if they are present in at least x% of the total number of replicates. In this example, a core taxa must account for 0.02% of the total reads for the entire otu table and be present in at least 50% of sites.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/abundance_and_occupancy_core.html">abundance_and_occupancy_core</a></span><span class="op">(</span><span class="va">arabidopsis</span>, occupancy_core <span class="op">=</span> <span class="fl">0.5</span>, prop_reads <span class="op">=</span> <span class="fl">0.0002</span><span class="op">)</span></code></pre></div>
<p><strong>We encourage collaboration – please submit a PR or issue if you’d like to see more methods included in the package!</strong></p>
</div>
</div>
<div id="all-methods" class="section level1">
<h1 class="hasAnchor">
<a href="#all-methods" class="anchor"></a>All Methods</h1>
<p>Rather than look at each method in isolation, we can use the main function of the package: <code>combine_methods</code>. This function prints your taxa table and a <code>logical</code> column for each method if that taxa is included in the column’s method. It also adds the class <code>core_methods</code> to your dataset so it can be used in subsequent plotting:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="op">(</span>
<span class="va">combine_arab</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/core_methods.html">core_methods</a></span><span class="op">(</span><span class="va">arabidopsis</span><span class="op">)</span>
<span class="op">)</span></code></pre></div>
<pre><code>## # A tibble: 59,560 x 6
##    X            Mean Variance    CV name                                   value
##    &lt;chr&gt;       &lt;dbl&gt;    &lt;dbl&gt; &lt;dbl&gt; &lt;fct&gt;                                  &lt;dbl&gt;
##  1 OTU_100   0.368    0.707   1.92  Summation of Sequence Reads                0
##  2 OTU_100   0.368    0.707   1.92  Proportion of Sequence Reads and Repl…     0
##  3 OTU_100   0.368    0.707   1.92  Hard Cut Off                               0
##  4 OTU_100   0.368    0.707   1.92  Proportion of Sequence Replicates          0
##  5 OTU_1000  0.00694  0.00692 0.997 Summation of Sequence Reads                0
##  6 OTU_1000  0.00694  0.00692 0.997 Proportion of Sequence Reads and Repl…     0
##  7 OTU_1000  0.00694  0.00692 0.997 Hard Cut Off                               0
##  8 OTU_1000  0.00694  0.00692 0.997 Proportion of Sequence Replicates          0
##  9 OTU_10000 0.698    5.85    8.38  Summation of Sequence Reads                0
## 10 OTU_10000 0.698    5.85    8.38  Proportion of Sequence Reads and Repl…     0
## # … with 59,550 more rows</code></pre>
</div>
<div id="plot" class="section level1">
<h1 class="hasAnchor">
<a href="#plot" class="anchor"></a>Plot</h1>
<div id="histo-bins" class="section level2">
<h2 class="hasAnchor">
<a href="#histo-bins" class="anchor"></a>Histo-bins</h2>
<p>This function takes on the output of <code>core_methods</code>. This plot is a hex bin plot to visually inspect the taxa that each method includes and the number of counts for each of those taxa. The function has one neccessary argument, <code>combined_otu_data</code>, but you can include the low and high colors for hex fills as well as a more informative legend title:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="../reference/core_plots.html">core_plots</a></span><span class="op">(</span><span class="va">combine_arab</span>, high<span class="op">=</span> <span class="st">"#1B301E"</span>, low <span class="op">=</span> <span class="st">"#E1E4E1"</span>, legend_title <span class="op">=</span> <span class="st">"Arabidopsis Taxa Count"</span><span class="op">)</span></code></pre></div>
<p><img src="analysis_and_plots_files/figure-html/unnamed-chunk-10-1.png" width="864"></p>
</div>
<div id="venn-diagram" class="section level2">
<h2 class="hasAnchor">
<a href="#venn-diagram" class="anchor"></a>Venn Diagram</h2>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">combine_arab</span> <span class="op">%&gt;%</span> <span class="fu"><a href="../reference/core_venn.html">core_venn</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="analysis_and_plots_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Maya Gans, Gordon Custer.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
